dcm4che-core/src/main/java/org/dcm4che3/io/stream/CloseableImageInputStreamAdapter.java

```diff
@@ -0,0 +1,79 @@
+//
+///////////////////////////////////////////////////////////////
+//                C O P Y R I G H T  (c) 2019                //
+//        Agfa HealthCare N.V. and/or its affiliates         //
+//                    All Rights Reserved                    //
+///////////////////////////////////////////////////////////////
+//                                                           //
+//       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF      //
+//        Agfa HealthCare N.V. and/or its affiliates.        //
+//      The copyright notice above does not evidence any     //
+//     actual or intended publication of such source code.   //
+//                                                           //
+///////////////////////////////////////////////////////////////
+//
+package org.dcm4che3.io.stream;
+
+import javax.imageio.stream.ImageInputStream;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ *
```

Comment:
Would be good to add some Javadoc with a link to the existing adapter and explaining the differences.

Reply:
Good point:  I've actually forgetten the differences myself now :-/

Reply:
I added comments to DelegateImageInputStream and CloseableImageInputStreamAdapter.  I deprecated the ImageInputStreamAdapter as the same behaviour can be created using the other two adapters.